lib/vrl/stdlib/src/parse_regex_all.rs

```diff
@@ -183,18 +181,13 @@ mod tests {
                               "veg": "carrots"},
                              {"fruit": "peaches",
                               "veg": "peas"}])),
-            tdef: TypeDef::new()
-                .fallible()
-                .array_mapped::<(), TypeDef>(map![(): TypeDef::new()
-                                                  .object::<&str, Kind>(map! {
-                                                      "fruit": Kind::Bytes,
-                                                      "veg": Kind::Bytes,
-                                                      "0": Kind::Bytes | Kind::Null,
-                                                      "1": Kind::Bytes | Kind::Null,
-                                                      "2": Kind::Bytes | Kind::Null,
-                                                  })
-                                                  .add_null()
-            ]),
+            tdef: TypeDef::array(Collection::from_unknown(Kind::null().or_object(btreemap! {
+                    Field::from("fruit") => Kind::bytes(),
+                    Field::from("veg") => Kind::bytes(),
+                    Field::from("0") => Kind::bytes() | Kind::null(),
+                    Field::from("1") => Kind::bytes() | Kind::null(),
+                    Field::from("2") => Kind::bytes() | Kind::null(),
+                }))).fallible(),
```

Comment:
Nit, same for the `tdef`s below. Looks like we don't get `cargo fmt` inside the macro here:

```suggestion
            tdef: TypeDef::array(Collection::from_unknown(Kind::null().or_object(btreemap! {
                Field::from("fruit") => Kind::bytes(),
                Field::from("veg") => Kind::bytes(),
                Field::from("0") => Kind::bytes() | Kind::null(),
                Field::from("1") => Kind::bytes() | Kind::null(),
                Field::from("2") => Kind::bytes() | Kind::null(),
            }))).fallible(),
```